### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by smavros**
Added API Documentation ( C and Python )
- addressees #192

### CODE REVIEW

File: core/docs/API.md [hunk 1]
```diff
@@ -21,32 +21,315 @@ The API exposes functions for:
...
+| `IO_Write_System_Energy( State * , const char* file, int idx_image, int idx_chain )`           | `void`     |
+
+
+| Chain Energies                                                                                 | Return     |
+| ---------------------------------------------------------------------------------------------- | ---------- |
+| `IO_Write_Chain_Energies( State * , const char* file, int idx_chain )`                         | `void`     |
+| `IO_Write_Chain_Energies_Interpolated( State * , const char* file, int idx_chain )`            | `void`     |
+
+
+### Exceptions
```
Comment (GPMueller) [inline core/docs/API.md]: you can remove this - the API functions should not throw exceptions (I know this is not correctly kept some times), as C does not have exceptions

File: core/docs/API.md [hunk 2]
```diff
@@ -21,32 +21,315 @@ The API exposes functions for:
 
 C API
```
Comment (GPMueller) [inline core/docs/API.md]: This is not needed. Instead, promote the Headings such as `State Management` or `Geometry` etc. to this level (i.e. with a line beneath.

File: core/docs/API_Python.md [hunk 3]
```diff
@@ -0,0 +1,160 @@
+SPIRIT API
+====================
+
+Python API
```
Comment (GPMueller) [inline core/docs/API_Python.md]: This is not needed. Instead, promote the other ### Headings to this level (as in the other file).

File: core/docs/API_Python.md [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+----------
+
+A new `State` can be created with
+
+```python
+    from spirit import state
+    with state.State("") as p_state:
+        pass
```
Comment (GPMueller) [inline core/docs/API_Python.md]: Add a comment  `# do something with p_state`

File: core/docs/API_Python.md [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+| --------------------------------------------------------------------------------------- | ------------- |
+| `Get_GNEB_Spring_Constant( p_state, c_spring, idx_image=-1, idx_chain=-1 )`             | `None`        |
+| `Get_GNEB_Climbing_Falling( p_state, climbing, falling, idx_image=-1, idx_chain=-1 )`   | `None`        |
+| `Get_GNEB_N_Iterations( p_state, idx_chain=-1 )`                                        | `Int`         |
+| `Get_GNEB_N_Iterations_Log( p_state, idx_chain=-1 )`                                    | `Int`         |
+| `Get_GNEB_N_Energy_Interpolations( p_state, idx_chain=-1 )`                             | `Int`         |
+
+### Simulation
+
+| Set Parameters                                                                                                            | Return     |
```
Comment (GPMueller) [inline core/docs/API_Python.md]: All your tables from here on have `Set Parameters` in the title - change this.

File: core/docs/API_Python.md [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+| `Get_Energy( p_state, idx_image=-1, idx_chain=-1 )`                      | `Float`    |
+| `Get_Energy_Array( p_state, energies, idx_image=-1, idx_chain=-1 )`      | `None`     |
+| `Update_Data( p_state, idx_image=-1, idx_chain=-1 )`                     | `None`     |
+| `Print_Energy_Array( p_state, idx_image=-1, idx_chain=-1 )`              | `None`     |
+
+### Transition
+
+| Set Parameters                                                               | Return   | Description                                                                        |
+| ---------------------------------------------------------------------------- | -------- | ---------------------------------------------------------------------------------- |
+| Homogeneous( p_state, idx_1, idx_2, idx_chain=-1 )                           | `None`   | Generate homogeneous transition between two images of a chain                      |
```
Comment (GPMueller) [inline core/docs/API_Python.md]: Code quotations are missing here

File: docs/UI-Qt.md [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+| Toggle Settings                                                   | <kbd>F2</kbd>                                        |
+| Toggle Plots                                                      | <kbd>F3</kbd>                                        |
+| Toggle Debug                                                      | <kbd>F4</kbd>                                        |
+| Toggle \"Dragging\" mode                                          | <kbd>F5</kbd>                                        |
+| Toggle large visualization                                        | <kbd>F10</kbd> / <kbd>Ctrl</kbd>+<kbd>F</kbd>        |
+| Toggle full-screen window                                         | <kbd>F11</kbd> / <kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>F</kbd> |
+| Screenshot of Visualization region                                | <kbd>F12</kbd> / <kbd>Home</kbd>                     |
+| Toggle OpenGL Visualization                                       | <kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>V</kbd>        |
+| Try to return focus to main UI (does not always work)             | <kbd>Esc</kbd>                                       |
```
Comment (GPMueller) [inline docs/UI-Qt.md]: remove the `(does not always work)`

File: docs/UI-Qt.md [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+| ------------------------------------------------ | :-----------------------------------------: |
+| Switch between images and chains                 | <kbd>&larr;</kbd> <kbd>&uarr;</kbd> <kbd>&rarr;</kbd> <kbd>&darr;</kbd> |
+| Cut image                                        | <kbd>Ctrl</kbd>+<kbd>X</kbd>                |
+| Copy image                                       | <kbd>Ctrl</kbd>+<kbd>C</kbd>                |
+| Paste image at current index                     | <kbd>Ctrl</kbd>+<kbd>V</kbd>                |
+| Insert left/right of current index               | <kbd>Ctrl</kbd>+<kbd>&larr;</kbd> / <kbd>&rarr;</kbd> | 
+| Delete image                                     | <kbd>Del</kbd>                              |
+
+
+<i>Note that some of the keybindings may only work correctly on US keyboard layout.</i><br>
```
Comment (GPMueller) [inline docs/UI-Qt.md]: Move this note to the top of this file.

**Comment by smavros** [comment]
Last commit
- addressees #197 #192

**Pull request merged**